adobe/aem-core-cif-components
# CIF-747 - Shopping cart popup

## Description

This PR adds the shopping cart component to the package. The MiniCart is a full client-side component, implemented using plain Javascript. It relies on a `CommerceApi` implementation that provides the connection to the commerce backend.

The shopping cart is created automatically when the page loads and its details are saved in a cookie called `cif.cart`. Once the "Add to cart" action is implemented the cart will be created when the first product is added.

How to test the feature:

1. Install the package and configure accordingly. 
2. Make sure you have your dispatcher instance configured properly and the requests to `http://localhost/magento/rest` and `http://localhost/magento/graphql` are forwarded correctly.
3. Open any page and click the "CartTrigger" icon from the header. The cart should slide from the left and be empty, showing the message "There are no items in your shopping cart"
4. Open your browser's developer console and add a product to the cart by calling the API function:

```javascript
window.CIF.MiniCart.addItem({sku:"VA03-KH-L",qty:"3"})
```
The cart should now display the product (without image, yet) and the cart trigger should display the number of items in the cart.
5. Click on the "kebab" menu next to the cart item and choose "Edit item" - the "edit" panel should slide from the left and allow you to change the quantity in the cart. Clicking on "Update" will close the edit panel and update the cart. 
6. Click on the "kebab" menu next to the cart item and choose "Remove item" - the item should be removed from the cart.

## Related Issue

CIF-747

## Motivation and Context

Provide a client-side reference component for a MiniCart.

## How Has This Been Tested?

Functional testing.

## Screenshots (if appropriate):

## Types of changes


## Checklist:
